geo-traits/src/dimension.rs

```diff
@@ -0,0 +1,34 @@
+/// The logical dimension of the geometry.
+///
+///
+#[allow(clippy::upper_case_acronyms)]
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub enum Dimension {
```

Comment:
I don't feel that strongly about this, but to me "Dimensions" means "you have multiple geometries, what is the dimension of each of them" and "Dimension" means "you have one geometry, what's its dimension?"

Reply:
Just to throw more fuel on the fire 😆:

I wonder if there's anything we can do to avoid confusion with https://github.com/georust/geo/blob/main/geo/src/algorithm/dimensions.rs#L25

Calling it something like `DimensionalOrdering` might be clearer, but it's such a mouthful!

It does make me realize that `YX` wouldn't be totally out of place in this enum, though I'm not sure if there's much utility in it.

Reply:
> I wonder if there's anything we can do to avoid confusion with [`main`/geo/src/algorithm/dimensions.rs#L25](https://github.com/georust/geo/blob/main/geo/src/algorithm/dimensions.rs?rgh-link-date=2024-10-12T19%3A03%3A12Z#L25)

Unfortunately `Dimension` is a very overloaded term. `HasDimensions` refers to the topological dimension, while the `Dimensions` here refers to what each topological dimension represents?

> It does make me realize that `YX` wouldn't be totally out of place in this enum, though I'm not sure if there's much utility in it.

I don't think `YX` would make sense. Instead you should use `XY` with a CRS that defines the two dimensions as `Y, X`, like EPSG:4326, which defines the coordinate ordering as latitude, then longitude.